src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,74 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { actions, currentTodoSelector } from '../../features/currentTodo';
+import { useAppSelector } from '../../app/store';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import { useDispatch } from 'react-redux';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  const currentTodo = useAppSelector(currentTodoSelector) as Todo;
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    getUser(currentTodo.userId)
+      .then(fetchedUser => {
+        setUser(fetchedUser);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, [currentTodo.userId]);
```

Comment:
Ensure that `currentTodo.userId` is always defined to prevent potential runtime errors when fetching user data. Consider adding a check or a default value to handle cases where `currentTodo.userId` might be undefined.